ui/src/dashboards/components/GraphOptionsCustomizableField.tsx

```diff
@@ -1,70 +1,186 @@
 import React, {PureComponent} from 'react'
+import {findDOMNode} from 'react-dom'
+import {
+  DragSourceSpec,
+  DropTargetConnector,
+  DragSourceMonitor,
+  DragSource,
+  DropTarget,
+  DragSourceConnector,
+} from 'react-dnd'
 
-import InputClickToEdit from 'src/shared/components/InputClickToEdit'
+const FieldType = 'field'
 
 interface Field {
   internalName: string
   displayName: string
   visible: boolean
+  order?: number
+  index?: number
 }
 
-interface Props {
+interface FieldSourceProps {
   internalName: string
   displayName: string
   visible: boolean
-  onFieldUpdate: (field: Field) => void
+  index: number
+  id: string
+  key: string
+  isDragging?: boolean
+  onFieldUpdate?: (field: Field) => void
+  connectDragSource?: any
+  connectDropTarget?: any
+  connectDragPreview?: any
```

Comment:
above there are a lot of optional props. Consider giving these props default values or making them not optional.

Reply:
I can't figure out how to do this.